File: src/previous/day67/InsertionSort.ts
export default function insertion_sort(arr: number[]): void {
  for (let i = 0; i < arr.length; i++) {
    for (let k = i; k > 0; k--) {
      if (arr[k] > arr[k - 1]) continue
      const tmp = arr[k]
      arr[k] = arr[k - 1]
      arr[k - 1] = tmp
    }
  }
}

File: go/day01/InsertionSort/InsertionSort.go
package insertion_sort

func InsertionSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		for k := i; k > 0; k-- {
			if arr[k-1] <= arr[k] {
				continue
			}

			tmp := arr[k]
			arr[k] = arr[k-1]
			arr[k-1] = tmp
		}
	}
}
